File: sentimental-credit/credit.py
from cs50 import get_int

if __name__ == "__main__":
    firstDigit = 0
    secondDigit = 0
    digits = 0

    doubleOdds = 0
    evens = 0

    valid = False

    cardNo = get_int("Enter the card number: ")

    while cardNo > 0:
        secondDigit = firstDigit
        firstDigit = cardNo % 10

        if digits % 2 == 0:
            evens += firstDigit
        else:
            multiple = 2 * firstDigit
            doubleOdds += (multiple // 10) + (multiple % 10)

        cardNo = cardNo // 10
        digits += 1

    if (evens + doubleOdds) % 10 == 0:
        valid = True

    firstTwo = (firstDigit * 10) + secondDigit

    if valid:
        if firstDigit == 4 and digits >= 13 and digits <= 16:
            print("VISA\n")
        elif firstTwo >= 51 and firstTwo <= 55 and digits == 16:
            print("MASTERCARD\n")
        elif firstTwo == 34 or firstTwo == 37 and digits == 15:
            print("AMEX\n")
        else:
            print("INVALID\n")
    else:
        print("INVALID\n")

File: credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    long card_no;
    string str[256];
    sprintf(str, "%ld", card_no);
    int sum = 0, card_no_length = str.length();
    bool secondChar = false;

    card_no = get_long("Enter the card number: ");

    while (card_no > 0)
    {
        for (int i = card_no_length - 1; i >= 0; i--)
        {
            int d = card_no[i] - '0';

            if(secondChar == true)
            {
                d = d * 2;
            }

            sum = d / 10;
            sum = d % 10;

            secondChar = !secondChar;
        }
    }

    if (sum % 10 == 0)
    {
        printf("%s, %s", str[0], str[1]);
    }
    else
    {
        printf("INVALID\n");
    }
}